### PR TITLE
GH-40216: [Python][CI][Packaging] Upload nightly wheels to main label of scientific-python-nightly-wheels channel

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
### Rationale for this change

Small follow-up on https://github.com/apache/arrow/pull/43862, correcting the `label` being used to upload the wheels. See https://github.com/apache/arrow/issues/40216#issuecomment-2325937999 for context.

* GitHub Issue: #40216

### CODE REVIEW

**Review by raulcd — APPROVED** [review]
Thanks! LGTM, I thought we might have wanted to mark them explicitly as dev but if that interferes with downloading from the index, this chaneg sounds good to me

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit -g wheel

File: dev/tasks/macros.jinja [hunk 1]
```diff
@@ -184,15 +184,13 @@ env:
 {% endmacro %}
 
 {%- macro github_upload_wheel_scientific_python(pattern) -%}
-  {%- if arrow.is_default_branch() -%}
```
Comment (pitrou) [inline dev/tasks/macros.jinja:187]: I suppose you'll restore this condition before merging the PR?

Comment (jorisvandenbossche) [reply dev/tasks/macros.jinja:187]: For sure! That's just to be able to test this here on the PR (I could maybe also just have merged it without testing, though)

**Comment by raulcd** [comment]
Some of the uploads to anaconda have failed but this seems more like a connection issues.  I will retry some of those jobs:
```
 Uploading file "scientific-python-nightly-wheels/pyarrow/18.0.0.dev255/pyarrow-18.0.0.dev255-cp310-cp310-macosx_12_0_x86_64.whl"

  0%|          | 0.00/29.8M [00:00<?, ?B/s]
  6%|▌         | 1.70M/29.8M [00:00<00:01, 17.8MB/s]
 30%|███       | 9.06M/29.8M [00:00<00:00, 52.6MB/s]
 53%|█████▎    | 15.8M/29.8M [00:00<00:00, 60.9MB/s]
 78%|███████▊  | 23.2M/29.8M [00:00<00:00, 67.2MB/s]
 99%|█████████▉| 29.6M/29.8M [00:00<00:00, 63.7MB/s]
29.8MB [00:00, 54.9MB/s]                            
Error: [ERROR] ('?: Undefined error ([POST] https://api.anaconda.org/commit/scientific-python-nightly-wheels/pyarrow/18.0.0.dev255/pyarrow-18.0.0.dev255-cp310-cp310-macosx_12_0_x86_64.whl -> 524)', 524)
Error: Process completed with exit code 1.
```

**Comment by jorisvandenbossche** [comment]
Now it is all green, so let's assume it was indeed a temporary connection issue

File: dev/tasks/macros.jinja (hunk 1 above)
Comment (jorisvandenbossche) [reply dev/tasks/macros.jinja]: Reverted the temp commit

**Pull request merged**